client/service/src/lib.rs

```diff
@@ -312,8 +318,8 @@ impl<TBl, TCl, TSc, TNetStatus, TNet, TTxPool, TOc> Spawn for
 		&self,
 		future: FutureObj<'static, ()>
 	) -> Result<(), SpawnError> {
-		self.task_manager.scheduler().unbounded_send((Box::pin(future), From::from("unnamed")))
-			.map_err(|_| SpawnError::shutdown())
+		self.task_manager.spawn_handle().spawn("unnamed", future);
```

Comment:
Not about pr changes, but this is going to be used more in future, can we do better identifying what and from where spawned  here?

Reply:
We can add an object that contains the task name in advance and implements `Spawn`, and pass it around.

For example, you'd do something like:

```rust
GrandPa::start(tasks_manager.spawn("grandpa"));
```